File: 1019-squares-of-a-sorted-array/1019-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        vector<int> ret;
        int size = nums.size();
        ret.resize(size);
        int l = 0;
        int r = size-1;
        int val = 0;
        
        for (int i = 0; i < size; i++) {
            if (fabs(nums[l]) >= fabs(nums[r])) {
                val = pow(nums[l], 2);
                l++;
            } else {
                val = pow(nums[r], 2);
                r--;
            }

            ret[size-1-i] = val;
        }


        return ret;
    }
};

File: 1019-squares-of-a-sorted-array/1019-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        left: int = 0
        right: int = 1
        ret: List[int] = []
        for i in range(len(nums)):
            left = i-1
            right = i
            if nums[i] >= 0:
                break
        while left >= 0 or right < len(nums):
            if left >= 0 and (right >= len(nums) or abs(nums[left]) <= abs(nums[right])):
                ret.append(pow(nums[left], 2))
                left-=1
            elif right < len(nums):
                ret.append(pow(nums[right], 2))
                right+=1
        return ret